controller/src/main/java/io/pravega/controller/store/stream/StreamCutComparison.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+/**
+ * Enumeration that encapsulates the result of comparison of two streamcuts.
+ * Two streamcuts are compared by comparing each segment and offset in each of the streamcuts with the corresponding 
+ * overlapping segments in the second streamcut. 
+ * A segment X from Streamcut SC1 is considered being ahead of segment Y in streamcut SC2 if either X is a successor of Y OR
+ * both X and Y are same segments and the offset in SC1 for X is greater than offset for X in SC2. 
+ */
+public enum StreamCutComparison {
+    // Indicates that for each segment S in streamcut SC1 there exists either the same segment with same or lesser offset 
```

Comment:
Is `S` a Segment? or a StreamCut? Can we explain this in a more simple way?

Reply:
Segment s in stream cut sc1.
Stream cut is described as offsets on different segments. It’s a map where keys are segments.